File: 268.missing-number.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#

# @lc code=start
from functools import reduce
from operator import xor


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        return len(nums) ^ reduce(xor, (i ^ n for i, n in enumerate(nums)))


# @lc code=end

File: 268.missing-number.rs
/*
 * @lc app=leetcode id=268 lang=rust
 *
 * [268] Missing Number
 */

// @lc code=start
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let n = nums.len() as i32;
        nums.into_iter()
            .enumerate()
            .fold(n, |acc, (i, x)| acc ^ (i as i32) ^ x)
    }
}
// @lc code=end
